src/inverse.jl

```diff
@@ -85,9 +85,22 @@ inverse(::typeof(inv)) = inv
 inverse(::typeof(adjoint)) = adjoint
 inverse(::typeof(transpose)) = transpose
 
+inverse(::typeof(!)) = !
 inverse(::typeof(+)) = +
 inverse(::typeof(-)) = -
 
+inverse(f::Base.Fix1{typeof(+)}) = Base.Fix2(-, f.x)
+inverse(f::Base.Fix2{typeof(+)}) = Base.Fix2(-, f.x)
+inverse(f::Base.Fix1{typeof(-)}) = Base.Fix1(-, f.x)
+inverse(f::Base.Fix2{typeof(-)}) = Base.Fix1(+, f.x)
+inverse(f::Base.Fix1{typeof(*)}) = iszero(f.x) ? throw(DomainError(f.x, "Cannot invert multiplication by zero")) : Base.Fix2(/, f.x)
```

Comment:
I think the inverse of `a * _` should be `a \ _`, not `_ / a`, so this will also work for complex numbers and matrices, resp. linear operators in general.